exporter.go

```diff
@@ -301,56 +311,84 @@ func (l *StatsDListener) handlePacket(packet []byte, e chan<- Events) {
 			continue
 		}
 
-		elements := strings.Split(line, ":")
+		elements := strings.SplitN(line, ":", 2)
 		if len(elements) < 2 {
 			networkStats.WithLabelValues("malformed_line").Inc()
 			log.Println("Bad line from StatsD:", line)
 			continue
 		}
 		metric := elements[0]
-		samples := elements[1:]
+		var samples []string
+		if strings.Contains(elements[1], "|#") {
+			// using datadog extensions, disable multi-metrics
+			samples = elements[1:]
+		} else {
+			samples = strings.Split(elements[1], ":")
+		}
 		for _, sample := range samples {
 			components := strings.Split(sample, "|")
 			samplingFactor := 1.0
-			if len(components) < 2 || len(components) > 3 {
+			if len(components) < 2 || len(components) > 4 {
 				networkStats.WithLabelValues("malformed_component").Inc()
 				log.Println("Bad component on line:", line)
 				continue
 			}
 			valueStr, statType := components[0], components[1]
+			labels := map[string]string{}
 			value, err := strconv.ParseFloat(valueStr, 64)
 			if err != nil {
 				log.Printf("Bad value %s on line: %s", valueStr, line)
 				networkStats.WithLabelValues("malformed_value").Inc()
 				continue
 			}
 
-			if len(components) == 3 {
-				if statType != "c" {
-					log.Println("Illegal sampling factor for non-counter metric on line", line)
-					networkStats.WithLabelValues("illegal_sample_factor").Inc()
-				}
-				samplingStr := components[2]
-				if samplingStr[0] != '@' {
-					log.Printf("Invalid sampling factor %s on line %s", samplingStr, line)
-					networkStats.WithLabelValues("invalid_sample_factor").Inc()
-					continue
-				}
-				samplingFactor, err = strconv.ParseFloat(samplingStr[1:], 64)
-				if err != nil {
-					log.Printf("Invalid sampling factor %s on line %s", samplingStr, line)
-					networkStats.WithLabelValues("invalid_sample_factor").Inc()
-					continue
-				}
-				if samplingFactor == 0 {
-					// This should never happen, but avoid division by zero if it does.
-					log.Printf("Invalid zero sampling factor %s on line %s, setting to 1", samplingStr, line)
-					samplingFactor = 1
+			if len(components) >= 3 {
+				for _, component := range components[2:] {
```

Comment:
Let's extract that into it's own function. The indentation level is already pretty high. Something like

```go
if len(components) >= 3 {
  values := parseDataDogLabels(components[2:])
}
```